generator/generator.go

```diff
@@ -1430,6 +1433,100 @@ func (g *generator) writeRawElement(indentLevel int, n parser.RawElement) (err e
 	return err
 }
 
+func (g *generator) writeScriptElement(indentLevel int, n parser.ScriptElement) (err error) {
+	if len(n.Attributes) == 0 {
+		// <div>
+		if _, err = g.w.WriteStringLiteral(indentLevel, `<script>`); err != nil {
+			return err
+		}
+	} else {
+		// <script></script>
+		if err = g.writeElementScript(indentLevel, n.Attributes); err != nil {
```

Comment:
This was confusing at first, but I understand now. This handles script components.